cpp/log/cert.cc

```diff
@@ -546,12 +566,314 @@ Cert::Status Cert::ExtensionStructure(int extension_nid,
       LOG(WARNING) << "Corrupt extension data";
       LOG_OPENSSL_ERRORS(WARNING);
     }
+
     return FALSE;
   }
 
   return TRUE;
 }
 
+bool Cert::IsRedactedHost(const std::string& hostname) {
+  // Split the hostname on '.' characters
+  std::vector<string> tokens = util::split(hostname, '.');
+
+  for (string str : tokens) {
+    if (str == "?") {
```

Comment:
I was trying to match exactly what the spec says as it refers to labels in the name. The RFC doesn't specify how to treat anything other than labels which are '?' e.g. "x?.example.com" doesn't fall into the definition of a redaction as I read it.